subsys/bluetooth/rpc/Kconfig

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config BT_RPC
+	bool "Bluetooth over RPC"
+	select NRF_RPC
+	select NRF_RPC_CBOR
+	help
+	  Enables Bluetooth serialization over RPC
+
+if BT_RPC
+
+choice BT_RPC_ROLE_CHOICE
+	prompt "Bluetooth over RPC role selection"
+	default BT_RPC_HOST if SOC_NRF5340_CPUNET
+	default BT_RPC_CLIENT
+	help
+	  Select the device role for Bluetooth over RPC. The default role is
+	  a client role.
+
+config BT_RPC_CLIENT
+	bool "Bluetooth over RPC Client"
+	select BT
+	help
+	  Bluetooth Low Energy API will use nRF PRC library to access host on
+	  the remote core.
+
+config BT_RPC_HOST
+	bool "Bluetooth over RPC Host"
+	select BT
+	help
+	  Bluetooth Low Energy API will use nRF PRC library to provide API for
+	  client on the remote core.
+
+endchoice
+
+config BT_RPC_INITIALIZE_NRF_RPC
+	bool "Automatically initialize nRF RPC library"
+	default y
+	help
+	  Initialize nRF RPC library during the system startup. Disabling this
+	  option allow user to initialize it in a different way.
+
+config CBKPROXY_OUT_SLOTS
+	int "Maximum number of RPC callback proxy output slots"
+	default 16
+	range 0 16383
+	help
+	  Output slot is used to save context for one callback that is called
+	  on local side, encoded, send and finally executed on the remote side.
+	  Each output slot takes 8 bytes of flash memory. Maximum number of
+	  input slots on the remote side should be the same as this value.
+
+
+config CBKPROXY_IN_SLOTS
+	int "Maximum number of RPC callback proxy input slots"
+	default 16
+	range 0 16383
+	help
+	  Input slot is used to save context for one callback that is called
+	  on remote side, encoded, send and finally executed on the local side.
+	  Each output slot takes 8 bytes of RAM memory. Maximum number of
+	  output slots on the remote side should be the same as this value.
+
+module = BT_RPC
+module-str = BLE over nRF RPC
+source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
+
+if BT_RPC_CLIENT
+
+# Overrides the default Bluetooth HCI driver
```

Comment:
Note, overriden defaults like this only works when `BT_RPC_CLIENT=y` from start.

If user selects `BT_RPC_CLIENT=y`  using menuconfig on a existing sample, for example `bluetooth/peripheral_lbs` then the value don't change, for example if `BT_HCI_BUS_TYPE` is `BT_RPMSG` before enabling `BT_RPC_CLIENT`, then it will still be `BT_RPMSG` and no `BT_NO_DRIVER`.

More on stucked symbols:
https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Reply:
I am aware that this is happening but `BT_HCI_BUS_TYPE` is an upstream config so I don't see a possibility to use prompt off like in the "stuck" symbols doc.

Reply:
I assume that having `BT_RPMSG=y` together with `BT_RPC_CLIENT=y` would be an error, correct ?

If that is the case then we should find a proper solution.

> I am aware that this is happening but BT_HCI_BUS_TYPE is an upstream config so I don't see a possibility to use prompt off like in the "stuck" symbols doc.

There is always the possibility to have a patch in `sdk-zephyr`, and it even sounds like something that would be upstreamable.
We improve upstream Zephyr daily, so that should not block us in this case, if you can make a proper upstreamable fix.

Reply:
@KAGA164 If this change can be moved upstream then that would be ideal. Supporting serialization of the API should be an acceptable reason to make these kinds of changes.

That could be made with the local modification in sdk-zephyr that we have now, but maybe changed to be something where an application could use an alternative implementation of the Host API, where the alternative implementation would be an RPC implementation in our case.

Reply:
Yes having those two configs enabled would be an error. What is the correct solution to this problem in your opinion ?
I did some tests where I modify the `BT_HCI_BUS_TYPE` config in our zephyr fork in this way:
```
choice BT_HCI_BUS_TYPE
	prompt "Bluetooth HCI driver" if !BT_RPC_CLIENT
```

or more generic by adding a new config for `BT_HCI_BUS_TYPE` prompt:
```
choice BT_HCI_BUS_TYPE
	prompt "Bluetooth HCI driver" if !CUSTOM_BT_HCI_BUS_TYPE
```

then configuration is switched correctly but when I build this example first using `west build -b nrf5340dk_nrf5340_cpuapp` then there is the `hci_rpmsg` child image in the build directory. Next I use a menuconfig and enable `BT_RPC` and `BT_HCI_BUS_TYPE` is set correctly to none and also `BT_RPMSG_NRF53` is disabled, then I run `west build` and the `rpc_host` image is added to the build directory but build system try also to build previous `hci_rpmsg` child image which is already in build directory from the previous build and building it fails because rpmsg configs are disabled. I suppose that we need some additional changes in the build system if we want to have possibility to enable `BT_RPC` from the menuconfig.

Reply:
This sounds like you should add RPC as a bus type, like this:
```
choice BT_HCI_BUS_TYPE
     config BT_RPC
           bool "Bluetooth over RPC"
endchoice
```

that would add `RPC` as a choice value inside `BT_HCI_BUS_TYPE` which is what you probably want in this case.